sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -468,12 +477,26 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
       s"Unsupported value type ${v.getClass.getName} ($v).")
   }
 
+  private def toAttributes(cols: Seq[String]): Seq[Attribute] = {
+    def resolve(colName: String) : Attribute = {
+      df.col(colName).named.toAttribute match {
+        case a: Attribute => a
+        case _ => throw new IllegalArgumentException(s"'$colName' is not a top level column.")
+      }
+    }
+    cols.map(resolve)
+  }
```

Comment:
@cloud-fan I noticed that `drop()` is resolving column names by `cols.map(name => df.resolve(name)` instead of `df.col(name)`.  The difference (other than the return type) is that `df.col()` will try to resolve using regex and adding metadata. Do you think we need to make this consistent?

Reply:
We should. To avoid breaking change, I think we should change `drop` to follow `fill` to make it more powerful.

Reply:
OK. I will do this in a separate PR which will address the issue with handling duplicate columns in `drop` when no columns are specified.